apps/public-docsite-v9/src/Concepts/BuildTimeStyles.stories.mdx

```diff
@@ -0,0 +1,222 @@
+      exclude: /node_modules/,
+      use: {
+        // or 'ts-loader'
+        loader: 'babel-loader',
+      },
+    },
+  ];
+}
+```
+
+**We recommend adding the Griffel webpack loader after babel-loader or ts-loader.**. The underlying babel transforms
+used by the webpack loader are configured by default to use the most basic language features. Therefore to avoid
+extra configuration or parsing errors, using the webpack loader after transpilation is the recommended approach.
+
+#### Typescript (and additional transpilation) support
+
+The Webpack loader [evaluate styles at runtime](#build-time-style-evaluation). For typescript
+it's necessary to install `@babel/preset-typescript` and configure it in the [babelOptions](https://github.com/microsoft/griffel/tree/main/packages/webpack-loader#configuring-babel-settings) of the loader. Similarly
+any other language features that might be required can also be configured here. While the order of loaders will
+ensure that the webpack loader will run on transpiled code (for a specific module), runtime evaluation can potentially require other
+modules that are not transpiled. Any additional language features can also be configured in the same way.
+
+### Babel
+
+The Babel preset is used internally by the Webpack loader. For applications the Webpack loader is appropriate
+since it is the most popular bundler.
+However, library developers might prefer to use the Babel preset since it is a better fit for tools such as [Rollup](https://rollupjs.org/guide/en/).
+
+```sh
+yarn add @griffel/babel-preset
+```
+
+Simply create a Babel configuration file such as `.babelrc` and add the preset.
+
+```js
+module.exports = {
+  presets: [
+    [
+      '@griffel',
+      {
+        // see https://github.com/microsoft/griffel/tree/main/packages/webpack-loader#configuring-babel-settings
+        babelOptions: {
+          presets: ['@babel/preset-typescript'],
+        },
+      },
+    ],
+  ],
+};
+```
+
+#### Typescript (and additional transpilation) support
+
+[Similar to above](#typescript-and-additional-transpilation-support)
+
+#### Importing from a third party package
+
+If Griffel is re-exported from a third party package, it's necessary to configure the preset to use a different
+module source. By default, the preset already handles imports from `@fluentui/react-components`.
+
+```js
+import { makeStyles } from 'custom-package';
+```
+
+```json
+{
+  "presets": [
+    [
+      "@griffel/babel",
+      {
+        "modules": [{ "moduleSource": "custom-package", "importName": "makeStyles" }]
+      }
+    ]
+  ]
+}
+```
+
+### What is being optimized
+
+> ⚠️ Style resolution only needs to happen on the initial render of a component. Therefore, without build time
+> optimization, the performance after the initial render is comparable. It is perfectly reasonable to follow the
+> quickstart guide and introduce build time optimization if/when it is required.
+
+Let's start with a simple example:
+
+```tsx
```

Comment:
**suggestion:** 

expand the example description a bit like following:


```tsx
import { makeStyles } from '@fluentui/react-components';

// 1.
const useStyles = makeStyles({
  root: { paddingLeft: '1px', display: 'flex' },
});

function Component() {
   // 2.
  const classes = useStyles();

  return <div className={classes.root} />;
```

What's happening in that example?

1. invocation of `makeStyles` creates a styling hook that will be used inside a component
2. `useStyles` hook call resolves styles which are injected into the document.

You can look at the image below which describes what work is done during style resolution. 

**💡 NOTE:  this work only happens once, during first render**.